rumqttc/src/eventloop.rs

```diff
@@ -227,7 +228,8 @@ impl EventLoop {
             },
             // We generate pings irrespective of network activity. This keeps the ping logic
             // simple. We can change this behavior in future if necessary (to prevent extra pings)
-            _ = self.keepalive_timeout.as_mut().unwrap() => {
+            _ = self.keepalive_timeout.as_mut().unwrap_or(&mut no_sleep),
+                if self.keepalive_timeout.is_some() && !self.mqtt_options.keep_alive.is_zero() => {
```

Comment:
i dont think `unwrap_or` is required here because of `self.keepalive_timeout.is_some()` guard but keeping that is also fine.

Reply:
I just double-checked this to be sure and it is required: The assignment expression is evaluated before the guard, so even though we check for `is_some()` and then skip the handling code that follows, it's not until after the system has called `unwrap()`. By making this `unwrap_or()`, we remove the possibility of a panic due to calling `unwrap()` on `None`.